src/sentry/sentry_apps/services/app/impl.py

```diff
@@ -61,20 +61,22 @@ def get_many(
     def find_app_components(self, *, app_id: int) -> list[RpcSentryAppComponent]:
         return [
             serialize_sentry_app_component(c)
-            for c in SentryAppComponent.objects.filter(sentry_app_id=app_id)
+            for c in SentryAppComponent.objects.using_replica().filter(sentry_app_id=app_id)
         ]
 
     def get_sentry_app_by_id(self, *, id: int) -> RpcSentryApp | None:
         try:
-            sentry_app = SentryApp.objects.get(id=id)
+            sentry_app = SentryApp.objects.using_replica().get(id=id)
```

Comment:
Have you considered using `get_from_cache()` on these id reads? It might be possible to get a good cache hit rate on these and skip postgres entirely most of the time.